lib/base.js

```diff
@@ -54,6 +59,72 @@ Base.prototype._checkStatusForError = function(statusCode, body) {
     }
 };
 
+Base.prototype.register = function({username, password, fields}, done) {
+    if (!username) {
+        throw new AirtableError('NOT_FOUND', "Missing parameter 'username' required for Base#register", 404);
+    }
+    if (!password) {
+        throw new AirtableError('NOT_FOUND', "Missing parameter 'password' required for Base#register", 404);
+    }
+    if (this._airtable._endpointUrl.includes('https://api.airtable.com')) {
+        throw new AirtableError('NOT_AUTHORIZED', 'Base#register cannot be used with api.airtable.com. Please configure this Airlock client to use an Airlock endpoint URL.', 403);
+    }
+    if (this._user !== null || this._getLocalStorage('user') !== null) {
```

Comment:
Currently when a login fails, you're setting up an empty object in local storage. This passes the logged in check, although the login actually failed!